dace/codegen/targets/fpga.py

```diff
@@ -420,13 +423,14 @@ def allocate_array(self, sdfg, dfg, state_id, node, function_stream,
                                                               bank)
                         else:
                             self._bank_assignments[(dataname, sdfg)] = None
+                        # TODO: Remove auto keyword
```

Comment:
Are you sure? This returns an `hlslib::ocl::Buffer<T, AccessType>`, it's not like we are preventing the compiler from doing optimizations here.

Reply:
Fixed